src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertPlaceholderToInterpolatedStringRefactoringProvider.cs

```diff
@@ -15,351 +15,303 @@
         where TArgumentSyntax : SyntaxNode
-        where TLiteralExpressionSyntax : SyntaxNode
         where TArgumentListExpressionSyntax : SyntaxNode
         where TInterpolationSyntax : SyntaxNode
     {
-
-        // Methods that are not string.Format but still should qualify to be replaced.
-        // Ex: Console.WriteLine("{0}", a) => Console.WriteLine($"{a}");
-        private static readonly ImmutableArray<(string typeName, ImmutableArray<string> methods)> s_compositeFormattedMethods = ImmutableArray.Create(
-            (typeof(Console).FullName!, ImmutableArray.Create(nameof(Console.Write), nameof(Console.WriteLine))),
-            (typeof(Debug).FullName!, ImmutableArray.Create(nameof(Debug.WriteLine), nameof(Debug.Print))),
-            (typeof(Trace).FullName!, ImmutableArray.Create(nameof(Trace.TraceError), nameof(Trace.TraceWarning), nameof(Trace.TraceInformation))),
-            (typeof(TraceSource).FullName!, ImmutableArray.Create(nameof(TraceSource.TraceInformation))));
-
-        protected abstract SyntaxNode GetInterpolatedString(string text);
+        protected abstract TExpressionSyntax ParseExpression(string text);
 
         public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
         {
-            var (document, textSpan, cancellationToken) = context;
+            var (document, span, cancellationToken) = context;
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
 
             var stringType = semanticModel.Compilation.GetSpecialType(SpecialType.System_String);
             if (stringType.IsErrorType())
-            {
                 return;
-            }
 
-            var stringInvocationMethods = CollectMethods(stringType, ImmutableArray.Create(nameof(string.Format)));
-            var compositeFormattedInvocationMethods = s_compositeFormattedMethods
-                .SelectMany(pair => CollectMethods(semanticModel.Compilation.GetTypeByMetadataName(pair.typeName), pair.methods))
-                .ToImmutableArray();
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
 
-            var allInvocationMethods = stringInvocationMethods.AddRange(compositeFormattedInvocationMethods);
+            var (invocation, placeholderArgument) = await TryFindInvocationAsync().ConfigureAwait(false);
+            if (invocation is null || placeholderArgument is null)
+                return;
 
-            if (allInvocationMethods.Length == 0)
-            {
+            var placeholderExpression = syntaxFacts.GetExpressionOfArgument(placeholderArgument);
+            var stringToken = placeholderExpression.GetFirstToken();
+
+            // don't offer if the string argument has errors in it, or if converting it to an interpolated string creates errors.
+            if (stringToken.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
```

Comment:
📝 I'd either remove this (expensive) test, or relocate it to inside the refactoring itself (i.e. offer the refactoring as long as the invocation binds, but it might be a NOP if there are errors in the application).

Reply:
seems.... problematic to offer to convert, but then have the refactoring do nothing?